static/script.js

```diff
@@ -9,6 +9,20 @@ function hash(path, handler) {
     });
 }
 
+/* --- VirusTotal --- */
+function MalShare_search(sha256, handler, interval=5000) {
+    $.get("/malshare/sha256/" + sha256, function(data, status) {
+        if (status == "success") {
+            handler(data);
+        } else {
+            setTimeout(function () {
+                MalShare_search(sha256, handler, interval); // retry until success
```

Comment:
there is a reason to retry? if it failed, i think it's better to just write an error instead of retrying